packages/core/lib/services/credential/service.js

```diff
@@ -79,11 +81,17 @@ class CredentialService {
     for (const signedVC of credentials) {
       vp.addCredential(signedVC);
     }
-    const kp = getKeypairFromDoc(keyDoc);
-    kp.signer = kp.signer();
-    const suite = getSuiteFromKeyDoc(kp);
     vp.setHolder(keyDoc.controller);
-    return vp.sign(suite, challenge, domain);
+    keyDoc.keypair = keyDocToKeypair(keyDoc, getDock());
+    return vp.sign(
+      {
+        ...keyDoc,
+        id: `${keyDoc.controller}#keys-1`, // HACK: make it work with SDK did resolution, this is an SDK limitation
```

Comment:
this wont work for all keydocs, you cant assume the id in wallet's case. you have to use keydoc.id which should previously set the correct value for the DID key